ext/vulnmdsrc/nvd/nvd.go

```diff
@@ -79,30 +79,38 @@ func (a *appender) BuildCache(datastore database.Datastore) error {
 		a.dataFeedHashes = make(map[string]string)
 	}
 
-	// Get data feeds.
+	//Get data feeds.
 	dataFeedReaders, dataFeedHashes, err := getDataFeeds(a.dataFeedHashes, a.localPath)
 	if err != nil {
 		return err
 	}
+
 	a.dataFeedHashes = dataFeedHashes
 
 	// Parse data feeds.
-	for dataFeedName, dataFeedReader := range dataFeedReaders {
-		var nvd nvd
-		if err = xml.NewDecoder(dataFeedReader).Decode(&nvd); err != nil {
-			log.WithError(err).WithField(logDataFeedName, dataFeedName).Error("could not decode NVD data feed")
-			return commonerr.ErrCouldNotParse
-		}
+	for dataFeedName, dataFileName := range dataFeedReaders {
+
+		if f, err := os.Open(dataFileName); err == nil {
+			//add check here
+			var nvd nvd
+			r := bufio.NewReader(f)
```

Comment:
This, among a few other changes you made, prevents us from streaming reads/writes and rather buffers them into memory. This will increase memory usage. While this doesn't consume a lot of memory, I'd rather we see if we can maintain the streaming property.